app/helpers/hosts_and_hostgroups_helper.rb

```diff
@@ -100,6 +100,60 @@ def realm_field(f, can_override = false, override = false)
             ).html_safe
   end
 
+  def host_puppet_environment_field(f, select_options = {}, html_options = {})
```

Comment:
Not for me to decide whether to change or not... 
(but maybe time to change that convention - `f` means nothing here)